src/frontends/paddle/src/input_model.cpp

```diff
@@ -259,21 +274,68 @@ std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelImpl::determine_cut_
-                                            std::istream* weight_stream) {
+void InputModel::InputModelImpl::load_consts(const std::basic_string<T>& folder_with_weights) {
+    for (const auto& item : m_var_places) {
+        const auto& var_desc = item.second->get_desc();
+        const auto& name = item.first;
+        if (ov::util::ends_with(name, std::string{"feed"}) || ov::util::ends_with(name, std::string{"fetch"}))
+            continue;
+        if (!var_desc.persistable())
+            continue;
+
+        FRONT_END_GENERAL_CHECK(var_desc.type().type() == ::paddle::framework::proto::VarType::LOD_TENSOR);
+        const auto& tensor = var_desc.type().lod_tensor().tensor();
+        Shape shape(tensor.dims().cbegin(), tensor.dims().cend());
+        const auto& type = get_ov_type(tensor.data_type());
+        const auto& data_length = shape_size(shape) * type.size();
+        std::vector<uint8_t> tensor_data(data_length);
+
+        bool read_succeed = false;
+        if (!folder_with_weights.empty()) {
+            std::ifstream is(get_const_path(folder_with_weights, name), std::ios::in | std::ifstream::binary);
+            FRONT_END_GENERAL_CHECK(is && is.is_open(), "Cannot open file for constant value.");
+            const size_t header_size = 16;
+            std::vector<char> header(header_size);
+            is.read(&header[0], header_size);
+
+            uint32_t dims_len = 0;
+            is.read(reinterpret_cast<char*>(&dims_len), 4);
+            std::vector<char> dims_struct(dims_len);
+            is.read(&dims_struct[0], dims_len);
+            read_succeed = read_tensor(is, reinterpret_cast<char*>(&tensor_data[0]), data_length);
+        } else {
+            FRONT_END_GENERAL_CHECK(false, "Folder with weights must be provided.");
+        }
+        FRONT_END_GENERAL_CHECK(read_succeed,
+                                "File containing constant with name ",
+                                name,
+                                " wasn't successfully read.");
+        auto const_node = opset7::Constant::create(type, shape, &tensor_data[0]);
+        const_node->set_friendly_name(name);
+        m_tensor_values[name] = const_node;
+    }
+}
+
+// load_consts with stream is compatible with new PaddlePaddle API.
+void InputModel::InputModelImpl::load_consts(std::istream* weight_stream) {
     for (const auto& item : m_var_places) {
         const auto& var_desc = item.second->get_desc();
         const auto& name = item.first;
         if (ov::util::ends_with(name, std::string{"feed"}) || ov::util::ends_with(name, std::string{"fetch"}))
             continue;
+
+        // var_desc.persistable() is used to mark node const value or not.
         if (!var_desc.persistable())
             continue;
 
         FRONT_END_GENERAL_CHECK(var_desc.type().type() == ::paddle::framework::proto::VarType::LOD_TENSOR);
+        FRONT_END_GENERAL_CHECK(weight_stream->peek() != EOF,
```

Comment:
Hi, @xczhai : just one question here, can't  this 'weight_stream' be nullptr? from 'get_model_path' [description](https://github.com/openvinotoolkit/openvino/blob/639235d09944f0a86503b0dda8b17e65a28214da/src/frontends/paddle/src/input_model.cpp#L202) looks like it can be nullptr(although this looks Strange and uncommon in real model case)

Reply:
Hi @liubo-intel,
- If loading model from disk, `weight_stream` can't be `nullptr` because its origin is from `std::ifstream` type in the method `InputModelImpl`.
- But your concern makes sense when loading from memory. So, I will add a `weight_stream == nullptr` check condition for this scenario.

Reply:
@liubo-intel Fixed.